packages/common-all/src/types/foundation.ts

```diff
@@ -45,6 +46,10 @@ export type DNodeProps<T = any, TCustom = any> = {
    * Node links (eg. backlinks, wikilinks, etc)
    */
   links: DLink[];
+  /**
+   * Anchors within the node (headings, block anchors)
+   */
+  anchors: { [index: string]: DNoteAnchorPositioned | undefined };
```

Comment:
when would it be useful for an anchor to be undefined?

Reply:
Lookups from the anchors have whatever type you give them, even though the key could be missing. i.e.:

```
const testMap: {[index: string]: string} = {};
const x: string = testMap["doesnt-exist"];
```

x has the type `string` even though the actual value can be (and is) `undefined`. So I added `undefined` to the type to reflect this. This seems to be a limitation of TypeScript.

Reply:
got it. makes sense